src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -415,9 +416,12 @@ Snapshot retrieveWorker(String sql, MapSqlParameterSource params) {
         }
 
         // Retrieve the Azure Storage Account associated with the snapshot.
-        resourceService
-            .getSnapshotStorageAccount(snapshot.getId())
-            .ifPresent(snapshot::storageAccountResource);
+        try {
+          snapshot.storageAccountResource(
+              resourceService.getSnapshotStorageAccount(snapshot.getId()));
+        } catch (StorageResourceNotFoundException ex) {
+          logger.debug(ex.getMessage());
```

Comment:
If you want to include the stack crawl too, you should pass in `ex`
```suggestion
          logger.debug(ex.getMessage(), ex);
```